### ISSUE TITLE
Rewrite documentation

### ISSUE THREAD

**Issue by oleksiyp** [envoyproxy/envoy#30341]
*Title*: please rewrite documentation

*Description*:
I used envoy proxy quite a lot. And love it. But key problem is usability of documentation.
Recently I returned back to configure it one more time and again struggling with it.
Suggestion is to rewrite documentation from scratch. Pick existing good documentation site and copy principles.
My suggestion is to use clean Gravitee docs as example, but you may check something else.

*Relevant Links*:
 * Gravitee docs: https://documentation.gravitee.io/apim/overview/readme

**Comment by kyessenov** [comment]
Thanks for the feedback. Please submit concrete proposals on improving the documentation. I agree it can always be improved, but not by rewriting it all.

**Comment by oleksiyp** [comment]
Right. So maybe worth to try change documentation website generator and search.

**Comment by kyessenov** [comment]
@phlax was looking into replacing sphinx, gitbook might be an option since it's a bit easier to author.

**Comment by phlax** [comment]
i think there are a few issues hre that are being confused

for sure the docs could do with improvement - the areas that spring to mind are:

- overall structure
- api generation
- formatting/grammar/spelling

switching the generator is pretty low down the list at this point tbh and doesnt solve any of those issues

fwiw i think rst is a good choice for this project - its annoying for the reasons that make it useful

i would be happy to review any PRs that deal with specific issues

**Comment by oleksiyp** [comment]
So what specific problems I had.
Menu too me is too overloaded to clearly navigate. Thus I use search. Found idle_timeout somehow which I needed. Then searched it. Got to proto reference docs and confirmed it is what I need. But missing examples and ways to go on upper level to understand how to use it in YAML, which place to put. So I was requiring ChatGPT and verify that.

**Comment by phlax** [comment]
> Menu too me is too overloaded to clearly navigate.

we are using the RTD theme and the navigation is ~configurable - ive been thinking about shifting off of the RTD theme but is very recognizable

> Thus I use search

search has never been great, and seems to have regressed in recent sphinx versions

again tho, this could well be as much to do with our setup/configuration

> But missing examples and ways to go on upper level

depending on where you are there is not currently a way to jump to anything relevant, bu there are ways of adding relevant context etc

like i said im happy to look at specific improvements, but i think changing the generator wouldnt (necessarily) resolve these issues

"rewriting" the docs is not a meagre undertaking - they are patchy and can be a bit "zombie" but they are the collected wisdom of many contributors, so rewriting would need to ensure no information was lost

**Comment by oleksiyp** [comment]
You decide. But I found this creates quite significant barrier(this time on tiny project). Also some time ago I tried to build Java controller providing configuration dynamically.  So yeah model is hard, nuanced.  Although yes it is possible to do everything that required, but requires high attention and time. So docs ideally should guide through, try to relax that complexity, dont see way relaxing model.

But generally envoy is best proxy ever, really cool thing itself, so happy if docs get better.

**Comment by phlax** [comment]
> ... so happy if docs get better.

me too, and would be happy to review contributions to that end
